include/pgtable.h

```diff
@@ -0,0 +1,256 @@
+/**
```

Comment:
Disclaimer: I am not a lawyer. There is a similar file in the UCX repository: ucx/src/ucs/datastruct/pgtable.h.
Do you believe that licenses are compatible?